File: 백준/Silver/5800. 성적 통계/성적 통계.c
#include <stdio.h>
#include <stdlib.h>

int compare(const void* a, const void* b){
   return (*(int*) b - (*(int*) a));
}

int main(void) { 
   int a, gap;
   scanf("%d", &a);

   for(int i = 0; i < a; i++){
      int b;
      scanf("%d", &b);
      
      int arr[b];
      for(int j = 0; j < b; j++){
         scanf("%d", &arr[j]);
      }

      qsort(arr, b, sizeof(int), compare);

      gap = -1;
      for(int k = 0; k < b - 1; k++){
         if(arr[k] - arr[k + 1] > gap){
            gap = arr[k] - arr[k + 1];
         }
      }

      printf("Class %d\nMax %d, Min %d, Largest gap %d\n", i + 1, arr[0], arr[b - 1], gap);
   }
   
   return 0;
}

File: 백준/Silver/5800. 성적 통계/성적 통계.py
a = int(input())

for i in range(a):
  gap = -1
  arr = list(map(int, input().split()))

  b = arr[0]
  arr = arr[1:]

  arr.sort()
  
  for j in range(b - 1, -1, -1):
    if arr[j] - arr[j - 1] > gap:
      gap = arr[j] - arr[j - 1]

  print(f'Class {i + 1}\nMax {max(arr)}, Min {min(arr)}, Largest gap {gap}')